var/spack/repos/builtin/packages/rust/package.py

```diff
@@ -71,13 +71,16 @@ class Rust(Package):
         default=False,
         description="Outputs code analysis that can be consumed by other tools",
     )
+    variant("cargo", default=True, description="The Rust package manager.")
     variant(
         "clippy",
-        default=True,
+        default=False,
         description="A bunch of lints to catch common mistakes and improve your Rust code.",
     )
     variant("docs", default=False, description="Build Rust documentation.")
-    variant("rustfmt", default=True, description="Formatting tool for Rust code.")
+    variant("rustdoc", default=True, description="Generate documentation for Rust projects.")
+    variant("rustfmt", default=False, description="Formatting tool for Rust code.")
+    variant("rust-analyzer", default=False, description="A Rust compiler front-end for IDEs")
```

Comment:
Is there any reason to have the defaults to `False`?[^1]

[^1]: I don't know `rust`, but in general I'd like to have tools for languages I use to code

Reply:
For rust development I totally agree, but if we include these additional tools into the default `rust` spec they're going to get built into all dependent cargo package environments meaning slower build times and larger rust installs for tools people may never use. Tools like `ripgrep`, `bat`, and `fd` are all unix utilities written in Rust but I doubt most people using them are rust developers who'd benefit from `rustfmt` or `rust-analyzer`.

I'd like to add a meta variant (`rust+dev`) that'd enable a bunch of development tools but I haven't figured out how to do that yet.

Reply:
> ... if we include these additional tools into the default rust spec they're going to get built into all dependent cargo package environments ...

They're going to get built, _unless_ they are disabled in environments etc. Anyhow, fine with me if we want to target a lightweight spec with the defaults. I guess one case is good for developers, the other targets more CI builds and the likes.